src/app/wallets/update-pending-payments.ts

```diff
@@ -0,0 +1,187 @@
+import { PaymentStatus } from "@domain/bitcoin/lightning"
+import { InconsistentDataError } from "@domain/errors"
+import { toLiabilitiesAccountId } from "@domain/ledger"
+import { FeeReimbursement } from "@domain/ledger/fee-reimbursement"
+import { LedgerService } from "@services/ledger"
+import { LndService } from "@services/lnd"
+import { LockService } from "@services/lock"
+import { PriceService } from "@services/price"
+
+export const updatePendingPayments = async ({
+  walletId,
+  logger,
+  lock,
+}: {
+  walletId: WalletId
+  logger: Logger
+  lock?: DistributedLock
+}): Promise<void | ApplicationError> => {
+  const liabilitiesAccountId = toLiabilitiesAccountId(walletId)
+
+  const lockService = LockService()
+  await lockService.lockWalletId({ walletId, logger, lock }, async () => {
```

Comment:
Changed to the `return` here https://github.com/GaloyMoney/galoy/pull/526/commits/0eb61e3282d8ceac7b5391597300f2901ecdf04c

> also, in the previous implementation only a lock is done when there is pending transactions, this is specially important because this is used in get balance an lock remains 10 seconds if something goes wrong

`updatePendingPayments` '`lock `' is optional here and only explicitly passed in when called together with `updatePendingInvoices` ([here](https://github.com/GaloyMoney/galoy/pull/526/files#diff-a9beb68fa5b83bb4910b21c35b84a1690884a90dd35209705698f1529b0c6c04R56) & [here](https://github.com/GaloyMoney/galoy/pull/526/files#diff-b5c296b20773b0ef9f3b15889c5f73b142201ae5aeb62961de05340f64e177f4R43)). Is this what you mean?

Reply:
what I mean is that you are locking the account even when the user has not pending payments,  this is the slower option (and this is used in get balance) and less safe for example when we do Promise.all in updatePending (if cant get the lock it will throw)

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/526/commits/2ef5e7d969157bbf501a983bb9fafef21f4054a2